srcopsmetrics/metrics.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+# Copyright (C) 2021 Dominik Tuchyna
+#
+# This file is part of thoth-station/mi - Meta-information Indicators.
+#
+# thoth-station/mi - Meta-information Indicators is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# thoth-station/mi - Meta-information Indicators is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with thoth-station/mi - Meta-information Indicators.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Metrics for MI."""
+
+import os
+from datetime import datetime
+from pathlib import Path
+
+import time
+import logging
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from github import Github
+
+from srcopsmetrics.entities.issue import Issue
+from srcopsmetrics.entities.pull_request import PullRequest
+from srcopsmetrics.utils import check_directory
+
+from typing import Dict
+
+_LOGGER = logging.getLogger(__name__)
+
+_GITHUB_ACCESS_TOKEN = os.getenv("GITHUB_ACCESS_TOKEN")
+
+
+class Metrics:
+    """Metrics used in MI."""
+
+    def __init__(self, repository: str, visuaize: bool = False):
+        """Initialize with collected knowledge."""
+        gh_repo = Github(login_or_token=_GITHUB_ACCESS_TOKEN, timeout=50).get_repo(repository)
+
+        self.repo_name = repository
+        self.prs = PullRequest(gh_repo).load_previous_knowledge(is_local=True)
+        self.issues = Issue(gh_repo).load_previous_knowledge(is_local=True)
+        self.visualize = visuaize
+
+    def get_aggregated_pull_requests(self) -> pd.DataFrame:
+        """Aggregate analysed data, calculate metrics from it and return DataFrame."""
```

Comment:
`and return DataFrame` is stated in the method output ` -> pd.DataFrame`

Moreover the name of the method, does not match the docstring, if I look for this method I expect it to just aggregate pull request not to also calculate metrics. Either rename the method or create another method for calculating metrics. WDYT?

Reply:
that's right, I've also changed the `plot_..` function to be named as `save_..`